packages/passwordless-auth-construct/src/services/cognito_user_service.test.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  AdminCreateUserCommandInput,
+  AdminDeleteUserAttributesCommandInput,
+  AdminUpdateUserAttributesCommandInput,
+  CognitoIdentityProviderClient,
+} from '@aws-sdk/client-cognito-identity-provider';
+import { beforeEach, describe, it, mock } from 'node:test';
+import { deepStrictEqual, rejects, strictEqual } from 'node:assert';
+import { CognitoUserService } from './cognito_user_service.js';
+
+/**
+ * A mock client for issuing send command successful
+ */
+class MockCognitoClient extends CognitoIdentityProviderClient {
+  send(): Promise<void> {
+    return Promise.resolve();
+  }
+}
+
+/**
+ * A mock client for issuing failing send command
+ */
+class MockCognitoClientFail extends CognitoIdentityProviderClient {
+  send(): Promise<void> {
+    return Promise.reject(new Error('error'));
+  }
+}
+
+void describe('Cognito User Service - Create User', async () => {
+  let mockCognitoClient: CognitoIdentityProviderClient;
+  void describe('Happy Path', async () => {
+    beforeEach(() => {
+      mockCognitoClient = new MockCognitoClient();
+    });
+
+    void it('should return successful for user created with email', async () => {
+      const sendMock = mock.method(mockCognitoClient, 'send');
+
+      strictEqual(sendMock.mock.callCount(), 0);
+
+      await new CognitoUserService(mockCognitoClient).createUser({
+        username: 'username',
+        userPoolId: 'userPoolId',
+        email: 'username@amazon.com',
+        region: 'us-west-2',
+      });
+
+      const expectedAttributes = {
+        MessageAction: 'SUPPRESS',
+        UserAttributes: [
+          {
+            Name: 'email',
+            Value: 'username@amazon.com',
+          },
+          {
+            Name: 'custom:passwordless_sign_up',
+            Value: '{"allowSignInAttempt":true,"deliveryMedium":"EMAIL"}',
+          },
+        ],
+        UserPoolId: 'userPoolId',
+        Username: 'username',
+      };
+
+      const actualCreateUserCommand = sendMock.mock.calls[0].arguments[0]
+        .input as unknown as AdminCreateUserCommandInput;
```

Comment:
```suggestion
        .input as AdminCreateUserCommandInput;
```
nit: I think unknown can be omitted

Reply:
Same for the other occurrences in this file